doc/projektivelho.md

```diff
@@ -54,3 +138,21 @@ classDiagram
     class MaterialGroup
     class MaterialCategory
 ```
+
+## Aineistojen metatiedotus (koodisto ja projektihierarkia)
+
+Projektivelhosta tuotavat inframallit sisältävät InfraModel-tiedostojen ulkopuolista metatietoa, josta 
+osa tallennetaan Geoviitteeseen. Aineiston projektihierarkia (projektijoukko,- projekti- ja 
+toimeksiantoviittaukset) tallennetaan siksi, että se voidaan näyttää operaattorille. Operaattorin on näiden 
+avulla mahdollista navigoida niiden kautta Projektivelhoon tutkimaan suurempaa kokonaisuutta aineiston ympäriltä. 
+Tämä informaatio on saatavilla vain projektivelhon kautta tuoduille aineistoille. Loput metatiedot (tiedoston 
+versio- ja koodistotiedot.) tallennetaan siksi, että siitä on mahdollisesti hyötyä tulevaisuudessa ja ne 
+vievät melko vähän tilaa.
```

Comment:
Vähän tämän kirjoittaminen sai kyllä miettimään että onko noiden "just because" -metatietojen tallentamisesta oikeasti mitään hyötyä meille 🤔 Mutta oh well

Reply:
Tiedoston versiotahan käytetään muistaakseni jo nyt siinä haussa/tallennuksessa, jotta nähdään onko kyseessä sama vai eri sisältö. PVDao, rivi 66+:
```
            on conflict (oid) do 
              update set
              ...
              where projektivelho.document.document_version <> :document_version

```

 Koodistot täytyy olla jotta koodien selkokieliset arvot voidaan nähdä ja kyllähän niistä ainaki joitain käytetäänkin (ainakin aineistoryhmä, dokumenttityyppi, dokumentin tila on ihan UI:lla esillä selkokielisinä). On siellä ehkä jotain mitä ei käytetä, mutta ei nuo pelkästään tulevaisuuden haaveina ole tallennettu.

+ Yksi syy miksi metatietoja kannattaa tallentaa vaikkei käyttäisikään, on se että jos jotain jää saamatta geoviiteeseen, noiden avulla voidaan debugata / löytää oikea rivi helpommin.